### PR TITLE
Add appveyor.yml

### PR DESCRIPTION
**Pull request by saurvs**

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      CHANNEL: nightly
+    - TARGET: i686-pc-windows-msvc
+      CHANNEL: nightly
+    - TARGET: x86_64-pc-windows-gnu
+      CHANNEL: nightly
+    - TARGET: x86_64-pc-windows-msvc
+      CHANNEL: nightly
+  
+matrix:
+  allow_failures:
```
Comment (suyash) [inline appveyor.yml]: Any particular reason to allow this?

Comment (saurvs) [reply appveyor.yml]: We won't be using any unstable Rust features, so it's probably better to not let that block PRs.

Comment (suyash) [reply appveyor.yml]: Yes, but why should that be an issue at this stage, why not leave it for when an issue occurs during implementation. For example in speed apart from tests I use this tool called gometalinter which is a combination of multiple linters. I have disabled and customized things to make tests pass, but at that stage it was justified. I guess why do this beforehand?

Comment (suyash) [reply appveyor.yml]: Also same re travis

Comment (saurvs) [reply appveyor.yml]: Yeah it's not too important either way since nightly failures are volatile, I'll remove it for now.

**Review by suyash — APPROVED** [review]

**Pull request merged**